### ISSUE TITLE
Say something about preservation of document properties when p:with-input/@select is used

### ISSUE THREAD

**Issue by gimsieke** [xproc/3.0-steps#138]
As I commented [here](https://github.com/xproc/3.0-steps/pull/131#pullrequestreview-250221972) a bit off-topic, if a pipeline author uses `p:with-input/@select`, the input document’s `content-type`, `serialization` and maybe `base-uri` properties might not be preservable even if the step says it will preserve all document properties.

Consider this:

```xml
<p:identity name="text">
  <p:with-input port="source">
    <p:inline>&lt;doc/></p:inline>
  </p:with-input>
</p:identity>
<!-- → Text document -->
<p:identity name="parse">
  <p:with-input port="source" select="parse-xml-fragment(/)">
</p:identity>
<!-- → XML document -->
```
Likewise, `p:identity` can turn an XML document into a text document using an appropriate `p:with-input/@select`. 

This is not limited to `p:identity`. So I don’t think we can say for any step that it will preserve all document properties under all circumstances.

Should we say something about it in the [steps spec introduction](http://spec.xproc.org/master/head/steps/#introduction), after the bullet list below “Some aspects of documents are generally unchanged by steps:”?

**Comment by xml-project** [comment]
👍

**Comment by ndw** [comment]
Consensus: we need to clarify content-type, base-uri, serialization, etc.

**Comment by ndw** [comment]
Closed by #149
